activerecord/lib/active_record/type/integer.rb

```diff
@@ -38,18 +38,21 @@ def cast_value(value)
 
       def ensure_in_range(value)
         unless range.cover?(value)
-          raise RangeError, "#{value} is out of range for #{self.class} with limit #{limit || 4}"
+          raise RangeError, "#{value} is out of range for #{self.class} with limit #{limit_or_default}"
         end
       end
 
       def max_value
-        limit = self.limit || 4
-        1 << (limit * 8 - 1) # 8 bits per byte with one bit for sign
+        1 << (limit_or_default * 8 - 1) # 8 bits per byte with one bit for sign
       end
 
       def min_value
         -max_value
       end
+
+      def limit_or_default
+        limit || 4
```

Comment:
Yeah, I originally planned on just extracting a constant, but then I figured I could name it *and* DRY things up a little by extracting a method.

Any thoughts on a better method name, @vipulnsward?

As far as I can tell the "4" is for 4 byte storage size, which means a [MySQL `int`](https://ariejan.net/2009/08/20/once-and-for-all-rails-migrations-integer-limit-option/) or a [Postgres `integer`](http://www.postgresql.org/docs/9.4/static/datatype-numeric.html), as opposed to `smallint`, `bigint` etc. Guess a constant + a comment could capture some of that information.